src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -1577,6 +1595,66 @@ public override BoundNode VisitCall(BoundCall node)
             return null;
         }
 
+        private bool MethodReturnsTrueWhenNotNull(MethodSymbol method)
+        {
+            if (method.ReturnType.SpecialType != SpecialType.System_Boolean)
+            {
+                return false;
+            }
+
+            return method.Equals(compilation.GetWellKnownTypeMember(WellKnownMember.System_String_IsNullOrEmpty));
```

Comment:
Please add a `// PROTOTYPE` comment to replace this with checking annotations. Same for `MethodEnsuresTrueWhenExits`.